src/Tinfoil/Data/Hash.hs

```diff
@@ -0,0 +1,26 @@
+{-# LANGUAGE NoImplicitPrelude #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE DeriveGeneric #-}
+module Tinfoil.Data.Hash(
+    Hash(..)
+  , hexDigest
+) where
+
+import           Data.ByteString (ByteString)
+import qualified Data.ByteString.Base16 as Base16
+import qualified Data.Text.Encoding as T
+
+import           GHC.Generics (Generic)
+
+import           P
+
+-- | Binary representation of a hash.
+newtype Hash =
+  Hash {
+    unHash :: ByteString
+  } deriving (Eq, Show, Generic)
+
+instance NFData Hash
```

Comment:
It's not a problem if a data type is strict all the way down, but that may not be the case if you have a `Maybe` or a `[]` in it - so yeah, in this case it's fine.

But in general I think we should be using [deepseq-th](https://hackage.haskell.org/package/deepseq-th-0.1.0.4/docs/Control-DeepSeq-TH.html) for deriving `NFData` until we've got everything switched to 7.10.